src/test/java/org/jclouds/vcloud/director/v1_5/compute/strategy/VCloudDirectorComputeServiceLiveTest.java

```diff
@@ -0,0 +1,83 @@
+package org.jclouds.vcloud.director.v1_5.compute.strategy;
+
+import static java.util.logging.Logger.getAnonymousLogger;
+import static org.jclouds.Constants.PROPERTY_USER_THREADS;
+import static org.testng.Assert.assertTrue;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import org.jclouds.compute.ComputeService;
+import org.jclouds.compute.domain.ComputeMetadata;
+import org.jclouds.compute.domain.NodeMetadata;
+import org.jclouds.compute.domain.Template;
+import org.jclouds.compute.domain.TemplateBuilder;
+import org.jclouds.compute.internal.BaseComputeServiceContextLiveTest;
+import org.jclouds.sshj.config.SshjSshClientModule;
+import org.testng.annotations.Test;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Lists;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.inject.Key;
+import com.google.inject.Module;
+import com.google.inject.name.Names;
+
+@Test(groups = "live", testName = "VCloudDirectorComputeServiceLiveTest")
+public class VCloudDirectorComputeServiceLiveTest extends BaseComputeServiceContextLiveTest {
+
+   protected ComputeService client;
+   protected Template template;
+
+   public VCloudDirectorComputeServiceLiveTest() {
+      provider = "vcloud-director";
+   }
+
+   @Override
+   protected Module getSshModule() {
+      return new SshjSshClientModule();
+   }
+
+   @Override
+   protected void initializeContext() {
+      super.initializeContext();
+      client = view.getComputeService();
+   }
+
+   public void testConcurrentlyDestroyNodes() throws Exception {
+      final String group = "twins";
+      List<NodeMetadata> nodes = Lists.newArrayList();
+      List<ListenableFuture<Boolean>> futures = Lists.newArrayList();
+      ListeningExecutorService userExecutor = context.utils().injector()
+              .getInstance(Key.get(ListeningExecutorService.class, Names.named(PROPERTY_USER_THREADS)));
+
+      try {
+         template = buildTemplate(client.templateBuilder().imageNameMatches("cloudsoft-template"));
+         template.getOptions().inboundPorts(22);
+         nodes.addAll(client.createNodesInGroup(group, 2, template));
+      } finally {
+         for (final NodeMetadata node : nodes) {
+            ListenableFuture<Boolean> future = userExecutor.submit(new Callable<Void>() {
+               public Void call() throws Exception {
+                  getAnonymousLogger().info("Destroying node " + node.getId());
+                  client.destroyNode(node.getId());
+               }
+            });
+            futures.add(future);
+         }
+         Futures.allAsList(futures).get();
+         assertTrue(client.listNodesDetailsMatching(new Predicate<ComputeMetadata>() {
```

Comment:
A jclouds node is mapped on a vCD VM, listNodes (and listNodeDetailsMatching) will return the list of VMs. An empty vApp could not be destroyed as it was probably created by other processes/manually. wdyt?